src/sqlfluff/core/rules/std.py

```diff
@@ -3407,6 +3399,126 @@ def _eval(self, segment, **kwargs):
         return self.violation_buff or None
 
 
+class OrderByColumnInfo(NamedTuple):
+    """For L035, segment that ends an ORDER BY column and any order provided."""
+
+    separator: BaseSegment
+    order: Optional[str]  # One of 'ASC'/'DESC'/None
+
+
+@std_rule_set.document_fix_compatible
+@std_rule_set.register
+class Rule_L035(BaseCrawler):
```

Comment:
I named this `Rule_L035` because I noticed there is a 34 and a 36, but no 35. Is this okay?

Reply:
As per #680 can we rename to 37?

Reply:
Will do!